### DISCUSSION TITLE
backport bot

### DISCUSSION THREAD

**Discussion by asvetlov** [aio-libs/aiohttp#5060] [General]
@webknjaz FYI

I've disabled (suspended) patchback bot for the sake of Backport action.
I'm fine with any solution but the action leaves comments in PR if backport is succeeded or failed.
The last is super important, I'd like to see a message if automatic backport is not possible and I should do it manually.
This feature is missed in patchback; that's why I cannot use it. If the bot supports it -- I'm open to try and use it again.

Thanks for understanding.

**Comment by webknjaz** [comment]
@asvetlov currently, patchback reports its status via Checks API on the PR where the backport was requested. Example: https://github.com/aio-libs/aiohttp/pull/4993/checks?check_run_id=1259218345.

I do, however, have a feature request for posting comments too: https://github.com/sanitizers/patchback-github-app/issues/13. It's coming from the @Ansible-collections where I started battle-testing patchback initially. I just haven't gotten to implementing it. It'll require a bit of refactoring to co-exist with Checks seamlessly, though. That's why it's not there yet.

Let's revisit this once the feature is implemented. Also, if you have any additional feature requests, feel free to file them at https://github.com/sanitizers/patchback-github-app/issues/new too.

**Reply by webknjaz** [reply]
The FR has been implemented. Also, it seems like that backport action is rather broken now.

**Reply by webknjaz** [reply]
Based on the above, I've unsuspended patchback.

**Comment by asvetlov** [comment]
Disabling chronographer for particular request by adding a label, e. g. `bot:chronographer:skip` would be great :)
Also, regarding backport I can imagine two things:
1. Setup a custom per-project labels for created pull request, as the backport action supports: https://github.com/aio-libs/aiohttp/blob/master/.github/workflows/backport.yml#L37
2. Automerge if configured (or label provided) PR when it has all required check passed (e.g. backport PR is approved and required checks are green). This functionality can be provided by a separate github app, as I've configured autosquash + backport actions in separate workflows. 

The CPython bot can automerge, it saves time for core devs: after approving and merging the main PR I monitor it for 20-30 seconds to make sure that all required backport PRs are created (or something has failed, both events are reported as comments). After that I go to generated backport PRs and approve them without waiting for CI and pressing "merge" for backports. It is very handy, the bot merges 95% of backports without additional human intervention.

**Reply by webknjaz** [reply]
> Disabling chronographer for particular request by adding a label, e. g. `bot:chronographer:skip` would be great :)

Urgh... Right. I'll get to it one day. I believe there's a tracking issue for that in the repo already.

> 1. Setup a custom per-project labels for created pull request, as the backport action supports: [`master`/.github/workflows/backport.yml#L37](https://github.com/aio-libs/aiohttp/blob/master/.github/workflows/backport.yml#L37)

The tracking issue is https://github.com/sanitizers/patchback-github-app/issues/12.

> 2. Automerge if configured (or label provided) PR when it has all required check passed (e.g. backport PR is approved and required checks are green). This functionality can be provided by a separate github app, as I've configured autosquash + backport actions in separate workflows.
> 
> The CPython bot can automerge, it saves time for core devs: after approving and merging the main PR I monitor it for 20-30 seconds to make sure that all required backport PRs are created (or something has failed, both events are reported as comments). After that I go to generated backport PRs and approve them without waiting for CI and pressing "merge" for backports. It is very handy, the bot merges 95% of backports without additional human intervention.

Yeah, that's totally understandable. Too bad that [miss-islington is hitting its rate limits already](https://discuss.python.org/t/miss-islington-hits-github-api-rate-limits/5246?u=webknjaz).
You are right, it's better to have automerging set up separately, and is imperative for it to rely on human approvals: the native dependabot incarnation has removed automerges because blind merges make repos vulnerable to the supply chain attacks.

**Reply by webknjaz** [reply]
> Urgh... Right. I'll get to it one day. I believe there's a tracking issue for that in the repo already.

`bot:chronographer:skip` is supported now.

**Reply by webknjaz** [reply]
@Dreamsorcerer do you want to explore GHA based automerge setup for the backports?

**Reply by Dreamsorcerer** [reply]
Probably not. :P Going to try and finish off some other things.

I assume we would only want to automerge if the bot is successful? If a user has to fix conflicts, automerging without review would be an obvious way to insert malicious code.

**Reply by webknjaz** [reply]
Yep. The condition would be if the backport PR was created by a bot.
